### ISSUE TITLE
SC2086 unnecessary for `readonly` assignments

### ISSUE THREAD

**Issue by climbTheStairs** [koalaman/shellcheck#2816]
#### For bugs
- Rule Id: SC2086
- My shellcheck version (`shellcheck --version` or "online"): 0.8.0, online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet that shows the problem:

```sh
#!/bin/sh

a='b c=d'
readonly e=$a

echo "$e" # "b c=d"
echo "$c" # ""
```

#### Here's what shellcheck currently says:

```
In test line 4:
readonly e=$a
           ^-- SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean:
readonly e="$a"
```

#### Here's what I wanted or expected to see:

Nothing

**Comment by Ordoviz** [comment]
See #2443. shellcheck warns here because Ubuntu 20.04 ships with dash 0.5.10.2, where the quotes are necessary (I tested it). Since then the POSIX spec has been updated so that quotes are no longer necessary.

**Comment by climbTheStairs** [comment]
Just realized this also applies to SC2046:

Input:
```sh
#!/bin/sh
readonly a=$(ls)
echo "$a"
```
Output:
```
In test.sh line 2:
readonly a=$(ls)
           ^---^ SC2046 (warning): Quote this to prevent word splitting.

For more information:
  https://www.shellcheck.net/wiki/SC2046 -- Quote this to prevent word splitt...
```
